### PR TITLE
[Draft] FIPS tests added for gnutls library

### LINKED ISSUE #495: Test fips mode of gnutls
We're not testing FIPS mode of gnutls, we should

### PR DESCRIPTION
**Pull request by rcmadhankumar**
[CI:TOXENVS] fips

fixes #495

### CODE REVIEW

File: tests/test_fips.py [hunk 1]
```diff
@@ -138,3 +168,48 @@ def fips_mode_setup_check(container_per_test: ContainerData) -> None:
...
+    - build a container image using :py:const:`DOCKERFILE_GNUTLS`
+    - run the bundled binary compiled from :file:`tests/files/fips-gnutls-test.c` with
+      all FIPS digests and assert that it successfully calculates the message
+      digest
+    - rerun the same binary with non-FIPS digests and assert that this fails
+      with the expected error message.
+
+    """
+
+    expected_fips_gnutls_digests = [
```
Comment (dcermak) [inline tests/test_fips.py:189]: Please convert this into a dictionary where the keys are the digest names and the hashes are values

**Review by dcermak — CHANGES_REQUESTED** [review]
This currently fails on SP3 & SP4, can you please adjust your test code akin to https://github.com/SUSE/BCI-tests/pull/570/files#diff-01da190f62559c48b5a3f8f593d99fa717493ed8925796aef63d9f145efd2d40 so that it'll work on the CI?

File: bci_tester/fips.py [hunk 2]
```diff
@@ -64,6 +64,28 @@
...
+#: gnutls digests that are not FIPS compliant
+NONFIPS_GNUTLS_DIGESTS: Tuple[str] = (
+    "gostr341194",
+    "streebog-256",
+    "streebog-512",
+)
+
+#: FIPS compliant gnutls digests
+FIPS_GNUTLS_DIGESTS: Tuple[str] = (
+    "md5",
```
Comment (dcermak) [inline bci_tester/fips.py:76]: md5 is not a FIPS digest

**Review by alexandrevicenzi — CHANGES_REQUESTED** [review]
The commit history is messy.

It seems the code changed here is also changed in #564, which can create conflicts once one PR is merged.

Consider cleaning this PR.

File: tests/test_fips.py [hunk 3]
```diff
@@ -152,3 +186,57 @@ def fips_mode_setup_check(container_per_test: ContainerData) -> None:
...
+    rr = container_per_test.connection.check_output(
+        
+        "zypper --gpg-auto-import-keys -n ref && zypper -n in gcc gnutls gnutls-devel && zypper -n clean && "
+        "gcc -Og -g3 fips-test-gnutls.c -Wall -Wextra -Wpedantic -lgnutls -o fips-test-gnutls && "
+        "mv fips-test-gnutls /bin/fips-test-gnutls && "
+        "rpm -q libgnutls-devel"
+    )
+    
+    assert "failureeee" in rr
+
```
Comment (dcermak) [inline tests/test_fips.py:216]:
```suggestion
```

File: tests/test_fips.py [hunk 4]
```diff
@@ -152,3 +186,57 @@ def fips_mode_setup_check(container_per_test: ContainerData) -> None:
...
+    - run the bundled binary compiled from :file:`tests/files/fips-gnutls-test.c` with
+      all FIPS digests and assert that it successfully calculates the message
+      digest
+    - rerun the same binary with non-FIPS digests and assert that this fails
+      with the expected error message.
+
+    """
+
+    rr = container_per_test.connection.check_output(
+
```
Comment (dcermak) [inline tests/test_fips.py:208]:
```suggestion
```

**Comment by dcermak** [comment]
@rcmadhankumar why did you revert https://github.com/SUSE/BCI-tests/pull/574?

**Comment by rcmadhankumar** [comment]
> @rcmadhankumar why did you revert #574?

PR itself was buggy. i merged it without testing.

File: bci_tester/fips.py (hunk 2 above)
Comment (rcmadhankumar) [reply bci_tester/fips.py]: yes, moved to NON FIPS Digests.

File: tests/test_fips.py (hunk 3 above)
Comment (rcmadhankumar) [reply tests/test_fips.py]: code is outdated(this section was added for debugging)

File: tests/test_fips.py (hunk 4 above)
Comment (rcmadhankumar) [reply tests/test_fips.py]: code is outdated(this section was added for debugging)

File: tests/test_fips.py (hunk 1 above)
Comment (rcmadhankumar) [reply tests/test_fips.py]: converted as dictionary.

**Comment by rcmadhankumar** [comment]
created a clean PR: https://github.com/SUSE/BCI-tests/pull/578

**Comment by alexandrevicenzi** [comment]
@rcmadhankumar if you update the branch (usually with -f if you change history) for a given PR, the PR gets updated. No need to create multiple PRs. It just confuses what I'm supposed to review.

**Pull request closed without merging**